lib/curlybars/method_whitelist.rb

```diff
@@ -100,6 +100,17 @@ def allow_methods(*methods_without_type, **methods_with_type, &contextual_block)
       define_method(:allows_method?) do |method|
         allowed_methods.include?(method)
       end
+
+      define_method(:as_json) do |*args|
+        return @__as_json if defined?(@__as_json)
+
+        @__as_json ||= allowed_methods.each_with_object({}) do |method, hash|
+          unless self.method(method).arity > 0
+            value = send(method)
+            hash[method] = value.instance_of?(self.class) && value == self ? "[circular reference]" : value.as_json
```

Comment:
I think [`#equal?`](https://devdocs.io/ruby~3.2/basicobject#method-i-equal-3F) is the method to use here:
```suggestion
            hash[method] = value.equal?(self) ? "[circular reference]" : value.as_json
```

Reply:
From the documentation seems to be a better way of checking it indeed.
Just committed your suggestion 👍🏼